miden-tx/src/error.rs

```diff
@@ -138,3 +138,152 @@ impl fmt::Display for DataStoreError {
+    (ERR_NOTE_INVALID_INPUTS, "Input note can not have empty inputs, procedure was likely called from the wrong context"),
+    (ERR_NOTE_TOO_MANY_ASSETS, "Note's asset must fit in a u32"),
+    (ERR_VAULT_GET_BALANCE_WRONG_ASSET_TYPE, "The get_balance procedure can be called only with a fungible faucet"),
+    (ERR_VAULT_HAS_NON_FUNGIBLE_WRONG_ACCOUNT_TYPE, "The has_non_fungible_asset procedure can be called only with a non-fungible faucet"),
+    (ERR_VAULT_FUNGIBLE_MAX_AMOUNT_EXCEEDED, "Adding the fungible asset would exceed the max_amount"),
+    (ERR_VAULT_ADD_FUNGIBLE_ASSET_MISMATCH, "Decorator value did not match the assert commitment"),
+    (ERR_VAULT_NON_FUNGIBLE_ALREADY_EXISTED, "The non-fungible asset already existed, can not be added again"),
+    (ERR_VAULT_FUNGIBLE_AMOUNT_UNDERFLOW, "Removing the fungible asset would have current amount being negative"),
+    (ERR_VAULT_REMOVE_FUNGIBLE_ASSET_MISMATCH, "Data provided via decorator did not match the commitment"),
+    (ERR_VAULT_NON_FUNGIBLE_MISSING_ASSET, "Removing inexisting non-fungible asset"),
+    (ERR_FUNGIBLE_ASSET_FORMAT_POSITION_ONE_MUST_BE_ZERO, "The felt at position 1 must be zero"),
+    (ERR_ASSET_FORMAT_POSITION_TWO_MUST_BE_ZERO, "The felt at position 2 must be zero"),
+    (ERR_FUNGIBLE_ASSET_FORMAT_POSITION_THREE_MUST_BE_ZERO, "The felt at position 3 must correspond to a fungible"),
+    (ERR_FUNGIBLE_ASSET_FORMAT_POSITION_ZERO_MUST_BE_ZERO, "The felt at position 0 must be within limit"),
+    (ERR_NON_FUNGIBLE_ASSET_FORMAT_POSITION_ONE_MUST_FUNGIBLE, "The felt at position 1 must be zero"),
+    (ERR_NON_FUNGIBLE_ASSET_HIGH_BIT_SET, "The felt at position 3 must be zero"),
+    (ERR_FUNGIBLE_ASSET_MISMATCH, "Fungible asset origin validation failed"),
+    (ERR_NON_FUNGIBLE_ASSET_MISMATCH, "Non-fungible asset origin validation failed"),
+    (ERR_ACCOUNT_NONCE_INCR_MUST_BE_U32, "The nonce increase must be a u32"),
+    (ERR_ACCOUNT_INSUFFICIENT_ONES, "Account id format is invalid, insufficient ones"),
+    (ERR_ACCOUNT_SET_CODE_ACCOUNT_MUST_BE_UPDATABLE, "Account must be updatable for it to be possible to update its code"),
+    (ERR_ACCOUNT_SEED_DIGEST_MISMATCH, "Account seed digest mismatch"),
+    (ERR_ACCOUNT_INVALID_POW, "Account pow is insufficient"),
+    (ERR_NOTE_DATA_MISMATCH, "Note's advice data does not match the expected commitment"),
+    (ERR_ASSET_NOT_FUNGIBLE_ID, "Can not build the fungible asset because provided id is not a fungible id"),
+    (ERR_ASSET_INVALID_AMOUNT, "Can not build the asset because amount exceeds the maximum"),
+    (ERR_ASSET_NOT_NON_FUNGIBLE_ID, "Can not build the non-fungible asset because provided id is not a non-fungible id"),
+    (ERR_INVALID_NOTE_TYPE, "Invalid note type"),
+    (ERR_NOTE_INVALID_TAG_PREFIX_FOR_TYPE, "The note's tag failed the most significant validation"),
+    (ERR_NOTE_INVALID_TAG_HIGH_BIT_SET, "The note's tag high bits must be set to zero"),
```

Comment:
Can you use numerals or digits instead of words for numbers? `zero` -> `0`. 

But in case you use a `Word`, we can use `ZERO`. Can you check everything to make it consistent?